test/shared_test_code.py

```diff
@@ -140,12 +140,23 @@ def check_pods_running(v1, pod_label_selector, count):
                 for pod in pods:
                     if pod.status.phase != "Running":
                         all_running = False
-                        break
-                if all_running: return True
+                        break    
+                if all_running:
+                    if pod_label_selector == BROKER_POD_LABEL_SELECTOR:
+                         return check_broker_pods_env_var(pods)
+                    else:
+                        return True
     print("Wrong number of pods [{}] found ... expected {}".format(
         pod_label_selector, count))
     return False
 
+def check_broker_pods_env_var(pods):
+    kubectl_cmd = get_kubectl_command()
+    for pod in pods:
+        if os.system('sudo {} exec -i {} -- /bin/bash -c "printenv | grep DEBUG_ECHO_DESCRIPTION=foo" | wc -l | grep -v 1"'.format(kubectl_cmd, pod.metadata.name)):
```

Comment:
Can you elaborate? Do you mean checking the debugEcho instance.properties?

Reply:
i wanted to test a user providing some property via the configuration and verify that it is passed all the way through.  so, verify that it is in the instance, etc.

Reply:
I see. Would it be enough to check the instance? Checking the broker seems one step further in validation

Reply:
@bfjelds, do the latest changes fit the scenario you were hoping for?

Reply:
not to close https://github.com/deislabs/akri/issues/179 as i understood it at the time.  to close that, we'd need to set properties in the config (not discoverydetails.description) and see that they were propogated through the system.

Reply:
I thought #179 was focusing on the instance metadata being propagated into the brokers not the `configuration.properties`. We have no support for propagating `configuration.properties` and that field should arguably be removed. I am not sure what the use case is behind it, since env vars could be directly set in the `brokerPodSpec` instead.

Reply:
Since properties are being removed in https://github.com/deislabs/akri/pull/290, does this check seem sufficient @bfjelds?

Reply:
i'm not sure how to feel about things at this point.  it seems like we no longer have a way for someone to add properties to a configuration.  sure, you can put anything into discoveryDetails, but we don't enable that (putting *anything* in) from helm.  so, conceptually we can remove properties ... but we haven't done anything to replace that concept.  i guess i don't love that.

as for this PR, how are discoveryDetails added to env vars?  is there an env var that is DISCOVERY_DETAILS and it has the serialized properties as the value?  i think we should explicitly test for what we expect rather than grep'ing for it from printenv.

Reply:
maybe we just call this a great improvement and see if anyone wants something different in the future :)

Reply:
discoveryDetails are not added as environment variables; rather, the `Device.properties` of the discovered device are. The name of the environment variable is different for each Discovery Handler.  For Debug Echo, the name of the environment variable is `DEBUG_ECHO_DESCRIPTION`.

I am not sure what you mean by " i think we should explicitly test for what we expect rather than grep'ing for it from printenv." By searching the env vars in the broker, this is verifying the correct env var was set.

Reply:
> discoveryDetails are not added as environment variables; rather, the Device.properties of the discovered device are. The name of the environment variable is different for each Discovery Handler. For Debug Echo, the name of the environment variable is DEBUG_ECHO_DESCRIPTION.

this is kind of the root of my issue.  previously, we intended to have a consistent (across all DHs) way for someone to communicate properties to a broker.  if they set a value in Configuration.Properties, it would end up in the broker.  it seems that is impossible.

> I am not sure what you mean by " i think we should explicitly test for what we expect rather than grep'ing for it from printenv." By searching the env vars in the broker, this is verifying the correct env var was set.

grepping the output of `printenv` isn't quite the same as testing for a specific env var's contents.  if someone changed the env var from DEBUG_ECHO_DESCRIPTION to MY_DEBUG_ECHO_DESCRIPTION, i think we'd want the test to catch it ... but the current grep test would not catch it.  i was suggesting that something more specific, like this, would be better: `echo $DEBUG_ECHO_DESCRIPTION | grep 'foo'`

Reply:
> if they set a value in Configuration.Properties, it would end up in the broker.

And the issue with just doing this in the broker pod spec is that it isnt verifiable? 

> like this, would be better: echo $DEBUG_ECHO_DESCRIPTION | grep 'foo'

How is that different from `printenv | grep DEBUG_ECHO_DESCRIPTION=foo`

Reply:
i think that `printenv | grep DEBUG_ECHO_DESCRIPTION=foo` would pass if this exists: `MY_DEBUG_ECHO_DESCRIPTION=foo`

Reply:
so, if someone created a PR that changed the env vars to prefix them with `AKRI_`, so that we ended up with `AKRI_DEBUG_ECHO_DESCRIPTION=foo` our test **should** fail ... but using `printenv | grep`, the tests would not fail.

Reply:
Weirdly, i couldnt get `kubectl exec -it jiria-linux-akri-debug-echo-ab76e6-pod -- /bin/sh -c "echo $DEBUG_ECHO_DESCRIPTION | grep bar | wc -l | grep -v 0"` to work. There may be some issue with permissions. instead i used regex for exact match

Reply:
might want to put a $ at the end of your grep to make sure that DEBUG_ECHO_DESCRIPTION=foob doesn't match

Reply:
The two description names are foo1 and foo2 so i am purposefully not doing an exact check